src/documents.c

```diff
@@ -200,3 +200,29 @@ register_document (const char *uri,
   doc_path = g_build_filename (documents_mountpoint, doc_id, basename, NULL);
   return g_filename_to_uri (doc_path, NULL, NULL);
 }
+
+char *
+get_real_path (const char *path,
+               const char *app_id,
+               GError **error)
+{
+  g_autofree char *doc_id = NULL;
+  gboolean ret = FALSE;
+  char *real_path = NULL;
+
+  if (app_id == NULL || *app_id == 0)
+    return g_strdup (path);
```

Comment:
Maybe the function should return `NULL` if there's an error?

Reply:
If the path is not a document path then it can be returned as-is because it's already "good".